aws/signer/v4/middleware.go

```diff
@@ -247,18 +247,18 @@ func (s *SignHTTPRequestMiddleware) HandleFinalize(ctx context.Context, in middl
 		return out, metadata, &SigningError{Err: fmt.Errorf("failed to retrieve credentials: %w", err)}
 	}
 
-	err = s.signer.SignHTTP(ctx, credentials, req.Request, payloadHash, signingName, signingRegion, sdk.NowTime(), s.addSignerOptions)
+	err = s.signer.SignHTTP(ctx, credentials, req.Request, payloadHash, signingName, signingRegion, sdk.NowTime(),
+		func(o *SignerOptions) {
```

Comment:
what do you think about this functional option returning an error?

Reply:
I think it made sense in LoadOptions case, but I'm not sure if I agree with signing having an error for an option setter function.

Reply:
we probably want to common pattern to be no error for these functional options, since their intention is to only modify configuration of a thing as it is invoked.  Potentially this could be applied to LoadOptions since, its functional options purpose are similar. The error value there may be a bit helpful for colliding options that aren't compatible. In the case of SignerOptions, the usage of the options is owned by the Signer, and not a separate entity like. The Signer it self can perform any validation as needed to determine that the options set are valid.